### PR TITLE
reduce amount of default logging on a failed request

### LINKED ISSUE #106: [Security] Headers content dumped in logs
Hello, in the event of a connection issue with the endpoint, the error message dumps the headers content in logs, which might lead to the exposure of the endpoint API key when headers are use for basic authentication and that logs are collected.

Would it be possible to add an option to make the inclusion of these headers conditional ? Thank you.

### PR DESCRIPTION
**Pull request by jsvd**
fixes #106

### CODE REVIEW

**Review by kaisecheng — COMMENTED** [review]
I left one comment for the variable name, otherwise LGTM

File: lib/logstash/outputs/http.rb [hunk 1]
```diff
@@ -252,16 +252,22 @@ def send_event(event, attempt)
...
-                :url => url,
-                :method => @http_method,
-                :body => body,
-                :headers => headers,
-                :message => exception.message,
-                :class => exception.class.name,
-                :backtrace => exception.backtrace,
-                :will_retry => will_retry
-    )
+    log_parameters = {
```
[the remark this answers is not in the document]
Comment (jsvd) [reply lib/logstash/outputs/http.rb]: well spotted!

**Comment by jsvd** [comment]
Thanks for the review @kaisecheng. Your comment made me realize two things:
* this code path wasn't actually being tested anywhere in the test suite
* the tests I added were covering the part of the code that handled the responses for requests that succeeded and not when `client.send` raised an exception

I added a new commit that fixes the variable name and also moved the tests to the right place. Now if the variable is not defined, the tests will fail.

File: spec/outputs/http_spec.rb [hunk 2]
```diff
@@ -165,6 +165,7 @@ def sinatra_run_wait(app, opts)
         it "should log a failure" do
           expect(subject).to have_received(:log_failure).with(any_args)
         end
+
```
Comment (jsvd) [inline spec/outputs/http_spec.rb:168]:
```suggestion
```

File: spec/outputs/http_spec.rb [hunk 3]
```diff
@@ -196,8 +196,44 @@ def sinatra_run_wait(app, opts)
         it "should make three total requests" do
           expect(subject).to have_received(:send_event).exactly(3).times
         end
+
+      end
+    end
+    context "on exception" do
+      before :each do
```
Comment (jsvd) [inline spec/outputs/http_spec.rb:203]:
```suggestion
      end
    end

    context "on exception" do
      before :each do
```

**Review by kaisecheng — APPROVED** [review]
LGTM 💯 test looks good as well.

**Pull request merged**